src/common/compat.cc

```diff
@@ -230,6 +232,26 @@ char *ceph_strerror_r(int errnum, char *buf, size_t buflen)
 #endif
 }
 
+int ceph_memset_s(void *dest, size_t destsz, int ch, size_t count)
+{
+  if (!dest || destsz > SIZE_MAX) {
+    return EINVAL;
+  }
+
+  if (destsz < count) {
+    ceph_memset_s(dest, destsz, ch, destsz);
```

Comment:
What's this doing?

Reply:
It's setting ``destsz`` bytes instead of ``count`` when the buffer is too small before returning ``EINVAL``, as per ``memset_s`` specs.

Reply:
I thought you were just planning to use `SecureZeroMemory` on Windows? Otherwise, wouldn't you technically need a memory barrier after clearing the buffer?

Reply:
Good point. To be honest, I'd much rather prefer using ``SecureZeroMemory`` but I thought you wanted a memset_s implementation https://github.com/ceph/ceph/pull/32777#discussion_r503970471.

The issue is that memset_s can write other characters as well, not just zero. I think the best thing to do is to define "ceph_zeromemory_s" and pick whatever's available out of ``memset_s``, ``SecureZeroMemory`` and ``explicit_bzero``. That, or my initial proposal of defining ``explicit_bzero`` if missing.